### ISSUE TITLE
TrustKit produces warnings during build

### ISSUE THREAD

**Issue by dodikk** [datatheorem/TrustKit#69]
I think, security related software is going to gain more trust if neither the compiler nor the static analysis reports any issues that might lead to exploits.

I would also recommend using `-wError` and `-wPedantic` to avoid such issues. 
For example, https://gist.github.com/dodikk/15d8df5f7a16d0f26204

```bash
/tmp/1/Carthage/Checkouts/TrustKit/TrustKit/Dependencies/domain_registry/private/trie_search.c:67:24: warning: no previous prototype for function 'FindNodeInRange' [-Wmissing-prototypes]
```

```bash
/tmp/1/Carthage/Checkouts/TrustKit/TrustKit/Dependencies/domain_registry/private/registry_search.c:256:25: warning: implicit conversion changes signedness: 'int' to 'size_t' (aka 'unsigned long') [-Wsign-conversion]
```

**Comment by nabla-c0d3** [comment]
Hello,
Thanks for the pull request! 
I agree and you can see that TrustKit "proper" does not report any compiler issues (even with the Analyzer). However, the files that you are mentioning here are from a Google library (https://github.com/pagespeed/domain-registry-provider). Hence, I did not want to "fork" the library, which is why I left the warnings in it.
It is true tho that these warnings should be addressed regardless. I can't merge your PR as it is now because it changes the coding style too, making it difficult to see in the diff what was actually fixed VS cosmetic changes. 
I do want to stay as close to the original files as possible, so if you want your changes to be merged, please open a PR that only fixes the warnings. Otherwise, I can do it myself, but I think it's cool for you if your contribution gets merged into the repo.
Let me know what you think.
Thanks!

**Comment by nabla-c0d3** [comment]
Fixed in c68b75d and 6596679.
